### ISSUE TITLE
ValueError: Invalid isoformat string

### ISSUE THREAD

**Issue by ghost** [bulletmark/pkglog#2]
On Arch Linux I'm seeing these `Invalid isoformat string` errors with the latest version from the [AUR package](https://aur.archlinux.org/packages/pkglog).

```sh
$ pacman -Q pkglog python
pkglog 1.27-2
python 3.11.3-2
```

```sh
$ /usr/bin/pkglog -V
1.27
```

```sh
$ /usr/bin/pkglog
Traceback (most recent call last):
  File "/usr/bin/pkglog", line 8, in <module>
    sys.exit(main())
             ^^^^^^
  File "/usr/lib/python3.11/site-packages/pkglog/pkglog.py", line 327, in main
    dt = logmod.get_time(line)
         ^^^^^^^^^^^^^^^^^^^^^
  File "/usr/lib/python3.11/site-packages/pkglog/parsers/pacman.py", line 25, in get_time
    dt = datetime.fromisoformat(f'{dts[1:23]}:{dts[23:-1]}')
         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
ValueError: Invalid isoformat string: '\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00:\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00[2022-09-14T07:53:11+0200'
```

If there's additional info needed, I'll be happy to add it here.

Regards

**Comment by bulletmark** [comment]
To be clear, you mean you have been using this ok for a while and this problem has started happening recently? Or you just installed it for the first time recently and see this error?

**Comment by ghost** [comment]
I have pkglog installed for a while. So definately not a first time issue. But I don't use it all the time, which is why I can't exactly pinpoint when this started to happen. It worked fine when I added a [comment](https://aur.archlinux.org/packages/pkglog#comment-919791) on the AUR page.

**Comment by bulletmark** [comment]
Seems your pacman log has at least one corrupted line but I guess the parser should be somewhat robust to such corruption so I have released new AUR version 1.28-1 which should address this.

**Comment by ghost** [comment]
I'll check my pacman.log. The new 1.28-1 does fix this. Thanks for your time and effort!
